src/globals.c

```diff
@@ -318,25 +318,47 @@ bool init_globals(void)
 		return false;
 	}
 
-/* Set configuration defaults based on options provided
-	at configure time.  */
+/* Set defaults with following order of preference:
+	1. Runtime flags
+	2. State dir configuration files
+	3. Configure time settings
+*/
+	if (!set_format_string(NULL)) {
 #ifdef FORMATID
-	set_format_string(FORMATID);
-#else
-	(void)set_format_string(NULL);
-#endif /* FORMATID */
+		/* Fallback to configure time format_string if other sources fail */
+		set_format_string(FORMATID);
+#endif
+}
 
+	if (set_version_url(NULL)) {
 #ifdef VERSIONURL
-	set_version_url(VERSIONURL);
+		/* Fallback to configure time version_url if other sources fail */
+		ret = set_version_url(VERSIONURL);
+		if (ret) {
+			printf("\nDefault version URL not found. Use the -v option instead.\n");
+			exit(EXIT_FAILURE);
+		}
 #else
-	set_version_url(NULL);
-#endif /* VERSIONURL */
+		/* We have no choice but to fail */
+		printf("\nDefault version URL not found. Use the -v option instead.\n");
```

Comment:
I would prefer not to see duplicate identical printf's like this... Can you move the printf and exit outside the ifdef block?